chiron/mcmc.py

```diff
@@ -112,103 +175,802 @@ def run(
+        current_nbr_list: Optional[PairsBase] = None,
+        calculate_current_potential: bool = True,
+    ) -> Tuple[SamplerState, ThermodynamicState, Optional[PairsBase]]:
+        """
+        Performs an individual MC step.
+
+        This will call the _propose  function which will be specific to the type of move.
 
         Parameters
         ----------
-        old_state : object
-            The state of the system before the move.
-        new_state : object
-            The state of the system after the move.
+        current_sampler_state : SamplerState, required
+            Current sampler state.
+        current_thermodynamic_state : ThermodynamicState, required
+            Current thermodynamic state.
+        current_nbr_list : Optional[PairsBase]
+            Neighbor list associated with the current state.
+        calculate_current_potential : bool, optional
+            Whether to calculate the current reduced potential. Default is True.
 
         Returns
         -------
-        float
-            Acceptance probability as a float.
+        sampler_state : SamplerState
+            The updated sampler state; if a move is rejected this will be unchanged.
+            Note, if the proposed move is rejected, the current PRNG key will be updated to ensure
+            that we are using a different random number for the next iteration.
+        thermodynamic_state : ThermodynamicState
+            The updated thermodynamic state; if a move is rejected this will be unchanged.
+            Note, many MC moves will not modify the thermodynamic state regardless of acceptance of the move.
+        nbr_list: PairsBase, optional
+            The updated neighbor/pair list. If a nbr_list is not set, this will be None.
+            If the move is rejected, this will correspond to the neighbor
+
         """
-        self._check_state_compatiblity(old_state, new_state)
-        old_system = self.system(old_state)
-        new_system = self.system(new_state)
 
-        energy_before_state_change = old_system.compute_energy(old_state.position)
-        energy_after_state_change = new_system.compute_energy(new_state.position)
-        # Implement the logic to compute the acceptance probability
+        # if this is the first time we are calling this function during this iteration
+        # we will need to calculate the reduced potential for the current state
+        # this is toggled by the calculate_current_potential flag
+        # otherwise, we can use the one that was saved from the last step, for efficiency
+        if calculate_current_potential:
+            current_reduced_pot = current_thermodynamic_state.get_reduced_potential(
+                current_sampler_state, current_nbr_list
+            )
+            # save the current_reduced_pot so we don't have to recalculate
+            # it on the next iteration if the move is rejected
+            self._current_reduced_pot = current_reduced_pot
```

Comment:
I'm not quite sure I understand.  are you suggesting renaming it "current_reduced_potential_energy"?  I don’t think this is a good idea, since it’s not JUST the reduce potential energy, as the reduced potential is defined as u = \beta [U(x) + p V(x) + \mu N(x)], and not just potential energy.